lib/vagrant/vagrantfile.rb

```diff
@@ -42,7 +42,7 @@ def initialize(loader, keys)
     # @return [Machine]
     def machine(name, provider, boxes, data_path, env)
       # Load the configuration for the machine
-      name = File.basename(name.to_s).delete(File.extname(name.to_s)).to_sym # Become a vmname if name from file path.
+      name = File.basename(name.to_s).sub(File.extname(name.to_s),'').to_sym # Become a vmname if name from file path.
```

Comment:
This change because of suspect that there is a possibility of deleting other unintended characters if it is a delete function.
I am sorry if my recognition was wrong.